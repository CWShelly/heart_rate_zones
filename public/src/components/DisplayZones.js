import React from 'react';
import Footer from './Footer'

const DisplayZones = (props)=>(

  <div className="displayContainer">

  {props.age > 0 && <div className="myZones"><p><span className="myZones-bold">Your age:</span> {props.age}.</p> <span className="myZones-bold">Your resting heart rate:</span> {props.rate} bpm.</div>}

  {props.mod_lower > 0 &&
    <div className="myZones">
     <p className="myZones-bold">For moderate intensity:</p>
     Stay between {props.mod_lower} and {props.vig_lower} bpm.
     </div>}

       {props.vig_lower > 0 &&  <div className="myZones"><p className="myZones-bold">For vigorous intensity:</p> Stay between {props.vig_lower} and {props.vig_higher} bpm.</div>}

       {props.max > 0 &&   <p className="myZones"> <span className="myZones-bold">Your max heart rate:</span> {props.max} bpm.</p>}
     {props.max > 0 && <Footer />}
  </div>
)




export default DisplayZones;
