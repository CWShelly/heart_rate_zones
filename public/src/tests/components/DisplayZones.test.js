import React from 'react';
import { shallow } from 'enzyme';
import DisplayZones from '../../components/DisplayZones';
import { emptyInfo, dummyInfo } from '../fixtures/heart_rate_info';


test('should correctly display DisplayZones page ',()=>{
  const wrapper = shallow(<DisplayZones />);
  expect(wrapper).toMatchSnapshot()
})

test('should handle DisplayZones data when nothing is added', ()=>{
    const wrapper = shallow(<DisplayZones message={emptyInfo.message}
      age={emptyInfo.age} mod_lower={emptyInfo.mod_lower}
      vig_higher={emptyInfo.vig_higher} max={emptyInfo.max} />);
        expect(wrapper).toMatchSnapshot();
})

test('should handle DisplayZones data when info is added', ()=>{
    const wrapper = shallow(<DisplayZones message={dummyInfo.message}
      age={dummyInfo.age} mod_lower={dummyInfo.mod_lower}
      vig_higher={dummyInfo.vig_higher} max={dummyInfo.max} />);
      expect(wrapper).toMatchSnapshot();
})
