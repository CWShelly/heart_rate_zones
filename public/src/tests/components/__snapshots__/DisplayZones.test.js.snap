// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly display DisplayZones page  1`] = `
<div
  className="displayContainer"
>
  <p
    className="myZones-bigger"
  />
</div>
`;

exports[`should handle DisplayZones data when info is added 1`] = `
<div
  className="displayContainer"
>
  <p
    className="myZones-bigger"
  >
    test message
  </p>
  <p
    className="myZones"
  >
    <span
      className="myZones-span"
    >
      Your age:
    </span>
     
    44
    . 
    <span
      className="myZones-span"
    >
      Your resting heart rate:
    </span>
     
  </p>
  <p
    className="myZones"
  >
    <span
      className="myZones-span"
    >
      For moderate intensity:
    </span>
     your recommended lower range: 
    100
    . Your recommended upper range: 
  </p>
  <p
    className="myZones"
  >
     
    <span
      className="myZones-span"
    >
      Your max heart rate: 
      176
    </span>
  </p>
</div>
`;

exports[`should handle DisplayZones data when nothing is added 1`] = `
<div
  className="displayContainer"
/>
`;
